wpilibc/src/main/native/cpp/PIDController.cpp

```diff
@@ -132,53 +132,86 @@ PIDController::~PIDController() {
  */
 void PIDController::Calculate() {
   bool enabled;
-  PIDSource* pidInput;
-  PIDOutput* pidOutput;
-
   {
-    std::lock_guard<wpi::mutex> lock(m_mutex);
-    pidInput = m_pidInput;
-    pidOutput = m_pidOutput;
+    std::lock_guard<wpi::mutex> lock(m_thisMutex);
+    if (m_pidInput == nullptr) return;
+    if (m_pidOutput == nullptr) return;
```

Comment:
Still a race.  If m_pidInput is set to nullptr while the lock is released, the later attempt to use it will crash.

Reply:
The only way m_pidInput can be modified is by the code in the constructor and `SetToleranceBuffer()`. For example,
```
m_filter = LinearDigitalFilter::MovingAverage(m_origSource, bufLength);          
m_pidInput = &m_filter;
```
Since m_filter is a value type, it's impossible for m_pidInput to go from non-nullptr to nullptr. I guess this should be checking m_origSource instead, since that was the original intent before LDF was added. Since that doesn't change, it can be moved back out of the lock.